src/uint/modular/constant_mod.rs

```diff
@@ -88,7 +88,17 @@ impl<MOD: ResidueParams<LIMBS>, const LIMBS: usize> Residue<MOD, LIMBS> {
     };
 
     /// Instantiates a new `Residue` that represents this `integer` mod `MOD`.
+    /// If the modulus represented by `MOD` is not odd, this function will panic; use [`new_checked`][`Residue::new_checked`] if you want to be able to detect an invalid modulus.
+    #[deprecated(
+        since = "0.5.3",
+        note = "This will return an `Option` in a future version to account for an invalid modulus, but for now will panic if this happens. Consider using `new_checked` until then."
+    )]
     pub const fn new(integer: &Uint<LIMBS>) -> Self {
```

Comment:
Unlike `DynResidue`, I don't think we need `new_checked` on `Residue`, and we definitely shouldn't deprecate it as having a const constructor is important.

Instead, I think we can lean on the compile-time checks in `impl_modulus!`, along with asserts to catch anyone trying to construct `ResidueParams` by hand.

Reply:
> having a const constructor is important

Very good point. Will update.

Reply:
Okay, fine, we can do that for now since we can't make invalid `ResidueParams` 100% unrepresentable yet without `generic_const_exprs`.

Can you at least add a TODO to remove `new_checked` when we can? We should really lean on compile time checks and the type system for this ASAP. Using runtime checks is a hack because we can't yet rule this case out with the type system.

Reply:
Agreed, and done.

Reply:
Hmm, now that I think about it, we can probably add a const-friendly assertion that the modulus is odd which will be evaluated at compile-time, and therefore obviate the need to have a runtime-checked version.

Let me make an inline suggestion.